examples/nng_test/pubsub.c

```diff
@@ -0,0 +1,141 @@
+/****************************************************************************
+ * apps/examples/nng_test/pubsub.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <nng/nng.h>
+#include <nng/protocol/pubsub0/pub.h>
+#include <nng/protocol/pubsub0/sub.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void fatal(FAR const char *func, int rv)
+{
+  fprintf(stderr, "%s: %s\n", func, nng_strerror(rv));
+}
+
+FAR char *date(void)
+{
+  time_t now = time(&now);
+  FAR struct tm *info = localtime(&now);
+  FAR char *text = asctime(info);
+  text[strlen(text) - 1] = '\0';
+  return text;
+}
+
+FAR void *client_thread(pthread_addr_t pvarg)
+{
+  nng_socket sock;
+  int rv;
+
+  sleep(2);
+
+  if ((rv = nng_sub0_open(&sock)) != 0)
+    {
+      fatal("nng_sub0_open", rv);
+      return NULL;
+    }
+
+  /* subscribe to everything (empty means all topics) */
+
+  if ((rv = nng_setopt(sock, NNG_OPT_SUB_SUBSCRIBE, "", 0)) != 0)
+    {
+      fatal("nng_setopt", rv);
+      return NULL;
```

Comment:
should we call `nng_sub0_close` or anything similar on failure path here and other places?

Reply:
I've took this example directly from nng, where there is no return statements in case of failure. Also I can not find close function. Beeing only an example I do not see it as a problem, althoug in some mission critical code, one must take all the precautions.

Reply:
if so, why not create example in netutils/nng/Makefile with the files from that folder.

Reply:
> if so, why not create example in netutils/nng/Makefile with the files from that folder to avoid the duplication?

Example is not actually a part of the library code, but is described on nng website.

Reply:
Ok.